p2p/host/autorelay/autorelay.go

```diff
@@ -135,6 +154,65 @@ func (ar *AutoRelay) background(ctx context.Context) {
 	}
 }
 
+func (ar *AutoRelay) refreshReservations(ctx context.Context, now time.Time) bool {
+	ar.mx.Lock()
+	if ar.status == network.ReachabilityPublic {
+		// we are public, forget about the relays, unprotect peers
+		for p := range ar.relays {
+			ar.host.ConnManager().Unprotect(p, autorelayTag)
+			delete(ar.relays, p)
+		}
+
+		ar.mx.Unlock()
+		return true
+	}
+
+	// find reservations about to expire and refresh them in parallel
+	var (
+		wg   sync.WaitGroup
+		fail int32
+	)
+
+	for p, rsvp := range ar.relays {
+		if rsvp == nil {
+			continue
+		}
+
+		if now.Add(rsvpExpirationSlack).Before(rsvp.Expiration) {
+			continue
+		}
+
```

Comment:
I must have missed this in my earlier review, but do we need a check to garbage collect expired reservations?

Reply:
I don't think so, we'll try to refresh them. Do you see any way to create garbage here?

Reply:
You're right, we delete the reservation if refreshing fails.